text/0155-external-modules.md

```diff
@@ -0,0 +1,106 @@
+This way they can reuse existing infrastructure for testing, publishing, security scanning etc.
+Also this will greatly simplify responsibility transfer between different maintainers if their priorities change. It also promotes the development
+and maintenance of a single instrumentation package for each instrumentation source, so that work isn't spread amongst multiple parallel solutions.
+
+Language SIGs are encouraged to provide a testing harness to verify that component adheres to OpenTelemetry semantic conventions
+and [recommendations for Otel instrumentations design](https://docs.google.com/document/d/1YNRCg9fdjJgZRs56vvf7rfFPk06mhp781sWHYypOaAk/edit#)
+when OpenTelemetry starts publishing them.
+
+A high volume of contrib contributions presents a burden for language maintainers. There are two suggestions for tackling this:
+
+- Create an "experimental" folder within contrib. The contents of this folder are not reviewed or maintained by language repository maintainers, but
+many of the other benefits of being within a contrib repository remain.
+- Add more approvers and maintainers, perhaps some who exclusively focus on submissions to contrib.
+
+### External components
+
+If component authors for whatever reason want to host their contribution outside the Otel contrib repository they are free to do so (though we
+encourage all contributions to go into contrib or core language repositories). Their submission for inclusion into OpenTelemetry Registry is still
+welcomed, subject to the same process described above.
+
+### Distribution
+
+Whenever OpenTelemetry components are published to any repository other than OpenTelemetry Registry (such as npm registry or Maven Central)
+only core and contrib components can be published under "opentelemetry" namespace.
+Native and external components are to be published under their own namespace.
+
+In case Otel SIG provides any kind of "all-in-one" instrumentation distribution (e.g. Java and .NET do) they should include
+only core and contrib packages into it.
+OpenTelemetry Registry should provide a way to easily obtain a list of these components.
+If possible SIG should provide a mechanism to include any external component during target application's build- or runtime.
+This may mean a separate language-specific component API that all components are encouraged to implement.
+
+## Trade-offs and mitigations
+
+* How easy it is to get merge permission in contrib repo?
+The harder it is, the larger is the maintenance burden on the core team.
+The easier it is, the more uncertainty there is about the quality of contributions.
+Can every language SIG decide this for themselves or should we decide together?
+
+## Open questions
+
+### Registry self-assessment form
+
+The exact list should be developed separately but at least component's author should declare that
+
+* It uses permissive OSS license
```

Comment:
Should we insist on no copyleft licenses, i.e. only allow licenses approved by CNCF?

Reply:
This is a good idea, I will add it clause.